pgmpy/models/MarkovModel.py

```diff
@@ -676,3 +676,30 @@ def get_partition_function(self):
             raise ValueError('Factor for all the random variables not defined.')
 
         return np.sum(factor.values)
+
+    def copy(self):
+        """
+        Returns a deep copy of this Markov Model.
+ 
+        Returns
+        -------
+        MarkovModel: Deep copy of this Markov model.
```

Comment:
A copy could either be shallow or a deep one. Explicitly specifying that this function will create a deep copy might be useful. This has no relation to `copy.deepcopy`.